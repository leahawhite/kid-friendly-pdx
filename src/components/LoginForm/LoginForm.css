.login-form {
  padding: 40px 20px;
  text-align: left;
}

.login-form input {
  border: 1px solid #ddd;
  display: block;
  font-size: 1.1em;
  margin-bottom: 10px;
  padding: 10px;
  text-align: left;
  width: 100%;
  border-radius: 4px;
}

.login-btn {
  order: 1px solid #aaa;
  background-color: #eee;
  font-size: 1.1em;
  border-radius: 4px;
  padding: 8px 12px;
}

h2 {
  font-size: 20px;
  margin: 0;
}

.signup {
  flex-direction: flex-end;
}