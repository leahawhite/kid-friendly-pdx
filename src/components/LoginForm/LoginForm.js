import React, { Component } from 'react';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service'
import './LoginForm.css';

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  constructor(props) {
    super(props)
    this.state = {
      error: null,
    }
  }

  handleSubmit = event => {
    event.preventDefault()
    this.setState({ error: null })
    const { email, password } = event.target

    AuthApiService.loginUser({
      email: email.value,
      password: password.value,
    })
      .then(res => {
        email.value = ''
        password.value = ''
        TokenService.saveAuthToken(res.authToken)
        this.props.onLoginSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    const { error } = this.state
    return (
      <form className="login-form" onSubmit={this.handleSubmit}>
        <div role='alert'>
          {error && <p className='red'>{error}</p>}
        </div>
        <div className="email">
          <label htmlFor="email">
            Email address
          </label>
          <input
            name="email"
            type="text"
            aria-label="email address"
            aria-required="true"
            required
            id="email"
            autoComplete="off" />
        </div>
        <div className="password">
          <label htmlFor="password">
            Password
          </label>
          <input
            name="password"
            type="password"
            aria-label="password"
            aria-required="true"
            required
            id="password"
            autoComplete="off" />
        </div>
        <button type="submit" className="login-btn">
          Log In
        </button>
          <p>Demo email: demo@demo.com</p>
          <p>Demo password: demo123</p>
      </form>
    )
  }
}