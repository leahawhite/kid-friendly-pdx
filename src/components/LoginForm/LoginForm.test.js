import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';

describe.only('Header component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders UI as expected', () => {
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
