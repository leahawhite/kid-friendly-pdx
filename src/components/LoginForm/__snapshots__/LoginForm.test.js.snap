// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders UI as expected 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    className="email"
  >
    <label
      htmlFor="email"
    >
      Email address
    </label>
    <input
      aria-label="email address"
      aria-required="true"
      id="email"
      name="email"
      required={true}
      type="text"
    />
  </div>
  <div
    className="password"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      aria-label="password"
      aria-required="true"
      id="password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    className="login-btn"
    type="submit"
  >
    Log In
  </button>
</form>
`;
